index.js

```diff
@@ -0,0 +1,209 @@
+const typeScriptSettings = {
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    ecmaVersion: 2018,
+    ecmaFeatures: {
+      jsx: true,
+    },
+    sourceType: 'module',
+    project: './tsconfig.json',
+  },
+  extends: [
+    'plugin:@typescript-eslint/recommended',
+    'prettier/@typescript-eslint',
+    'plugin:prettier/recommended',
```

Comment:
**Note:** The 'plugin:import/typescript' extend is also needed to import types out of `.d.ts` files (from node modules)

Reply:
I added `'plugin:import/typescript'`. This also enables the linting rule [import/no-cycle](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-cycle.md). This will cause a lot of issues with the `MubanTransitionController`, since there is a circle dependency by default (the component has a reference to the TransitionController and the TransitionController has a reference to the component).

So this is something we need to fix in the `MubanTransitionController`